runelite-client/src/main/java/net/runelite/client/plugins/microbot/qualityoflife/QoLPlugin.java

```diff
@@ -319,83 +425,147 @@ private void customAnvilOnClicked(MenuEntry event) {
-        executeWorkbenchActions = true;
+    private int getSmallestAngle(int x, int y) {
+        return ((y - x + HALF_ROTATION) % FULL_ROTATION) - HALF_ROTATION;
     }
 
     private void addMenuEntry(MenuEntryAdded event, String option, String target, Consumer<MenuEntry> callback) {
-        List<MenuEntry> entries = new LinkedList<>(Arrays.asList(Microbot.getClient().getMenuEntries()));
-
-        if (entries.stream().anyMatch(e -> e.getOption().equals(option) && e.getTarget().equals(target))) {
-            return;
-        }
-
-        Microbot.getClient().createMenuEntry(entries.size())
+        int index = Microbot.getClient().getMenuEntries().length;
+        Microbot.getClient().createMenuEntry(index)
                 .setOption(option)
                 .setTarget(target)
                 .setParam0(event.getActionParam0())
                 .setParam1(event.getActionParam1())
                 .setIdentifier(event.getIdentifier())
                 .setType(event.getMenuEntry().getType())
                 .onClick(callback);
-
     }
 
     private void addLoadoutMenuEntry(MenuEntryAdded event, String option, String target, Consumer<MenuEntry> callback) {
-        Microbot.getClient().createMenuEntry(1)
+        int index = Microbot.getClient().getMenuEntries().length - 1;
+        Microbot.getClient().createMenuEntry(index)
                 .setOption(option)
                 .setTarget(target)
                 .setParam0(event.getActionParam0())
                 .setParam1(event.getActionParam1())
                 .setIdentifier(event.getIdentifier())
                 .setType(event.getMenuEntry().getType())
                 .onClick(callback);
+    }
+
+    public void updateLastWinthertodtAction(WintertodtActions action) {
+        configManager.setConfiguration("QoL", "wintertodtActions", action);
+    }
+
+    public void updateWintertodtInterupted(boolean interupted) {
+        configManager.setConfiguration("QoL", "interrupted", interupted);
+    }
+
 
+    private void updateUiElements() {
+        try {
+            // Get the Field object for BRAND_ORANGE(Accent color)
+            Field accentColorField = ColorScheme.class.getDeclaredField("BRAND_ORANGE");
+            Class<?> pluginButton = Class.forName("net.runelite.client.plugins.config.PluginToggleButton");
+            Field onSwitcherPluginPanel = pluginButton.getDeclaredField("ON_SWITCHER");
+            onSwitcherPluginPanel.setAccessible(true);
+
+
+            FieldUtil.setFinalStatic(accentColorField, config.accentColor());
+
+            FieldUtil.setFinalStatic(onSwitcherPluginPanel, remapImage(SWITCHER_ON_IMG, config.toggleButtonColor()));
+            ConfigPlugin configPlugin = (ConfigPlugin) Microbot.getPluginManager().getPlugins().stream().filter((plugin) ->
+                    plugin instanceof ConfigPlugin).findAny().orElse(null);
+            if (configPlugin == null) {
+                throw new RuntimeException("Config Plugin not found");
```

Comment:
I think using Microbot.log might be better than throwing an exception. Atleast the user will see in the chatbox that something went wrong and can help us debug.